### PR TITLE
[half] Fix a bug in the half.hpp include file not enabling F16C intrinsics correctly when the architecture supports it.

### PR DESCRIPTION
**Pull request by darjankrijan**
Bug:
The immintrin.h header will not be included if ```HALF_ENABLE_F16C_INTRINSICS``` wasn't provided manually.
```HALF_ENABLE_F16C_INTRINSICS``` though will be set to ```__F16C__``` which compilers (GCC) provide if a proper ```-march``` flag is given, so this enables the intrinsics without having included the header above.
Compiler errors ensue because things like ```_mm_set_ss``` are not declared.

PR fixes this by putting the include directive after ```HALF_ENABLE_F16C_INTRINSICS``` being set to ```__F16C__``` which the compiler should set.

Issue mailed to upstream, bug is known but no bugfix release out yet.
Keep a look out for next release where this will hopefully be fixed.

Patches #94

### CODE REVIEW

File: half/PKGBUILD
Comment (acxz) [inline half/PKGBUILD:22]: Instead of just commenting can you link to the issue thread for this: https://github.com/acxz/pkgbuilds/issues/94
Feel free to keep your comment as well.

**Review by acxz — CHANGES_REQUESTED** [review]
Thanks for the fix @darjankrijan 
Is there anyway you can link to the upstream bug report and/or the upstream submitted patch?

I would like to get a link that refers back to upstream somehow regarding this before I merge this in.

**Comment by darjankrijan** [comment]
I don't have a link as I didn't open an issue on sourceforge because I don't have an account on that horrible site.

So I mailed the maintainer Christian Rau directly and got the reply that it is known and fixed internally, but not yet released.

**Comment by acxz** [comment]
> because I don't have an account on that horrible site

lol I feel that, thats a good enough reason for me lol.

**Review by acxz — APPROVED** [review]
LGTM

**Pull request merged**
